infra/faucet/src/ogn.js

```diff
@@ -28,8 +33,14 @@ class OgnDistributor {
       // Transfer NUM_TOKENS to the specified wallet.
       const value = this.token.toNaturalUnit(NUM_TOKENS)
       const contractAddress = this.token.contractAddress(networkId)
-      const receipt = await this.token.credit(networkId, wallet, value)
-      const txHash = receipt.transactionHash
+      const txHash = await this.token.credit(wallet, value)
+      const { status } = await this.token.waitForTxConfirmation(txHash, {
+        numBlocks: NumBlockConfirmation,
+        timeoutSec: ConfirmationTimeoutSec
+      })
+      if (status !== 'confirmed') {
```

Comment:
I'm not sure this needs to be dealt with now, but you might want to consider what happens in production if the tx takes longer than 2 minutes to get mined.  For instance, will this record the error, assume it won't get mined(but it does), then retry later creating multiple transactions?

Reply:
Currently error handling is poor on the OGN faucet. If transaction does not get mined within the timeout, there is no retry or rebroadcast or anything like that...
I added TODO - not a high priority since OGN faucet usage is extremely low.

Reply:
To be clear, my concern was if it **does** get mined, but the timeout still occurs.

Reply:
Yes, makes sense. I added TODo to handle return of either error or timeout from the call to token.waitForTxConfirmation. Really if it gets usage we'll need to improve the faucet. It does not handle neither concurrent transactions (will re-use same nonce)...